master/internal/scheduler/fitting.go

```diff
@@ -127,47 +119,32 @@ func findMultiSlotFits(
 		numSlots = append(numSlots, n)
 	}
 
-	if task.fittingRequirements.SingleAgent {
-		// For the single agent case, we want prioritize using the smallest
-		// available agents.
-		sort.Sort(numSlots)
-	} else {
-		// For the (potentially) distributed case, we want to use as few
-		// agents as possible, thus we prioritize the largest agents.
-		sort.Sort(sort.Reverse(numSlots))
-	}
+	// For the distributed case, we want to use as few
+	// agents as possible, thus we prioritize the largest agents.
+	sort.Sort(sort.Reverse(numSlots))
 
 	var candidateNumSlots int
 	for _, n := range numSlots {
 		if n == 0 {
 			continue
 		}
 
-		var maxSlots int
-		if task.fittingRequirements.SingleAgent {
-			maxSlots = n
-		} else {
-			agents := agentsByNumSlots[n]
-			maxSlots = len(agents) * n
-		}
-
-		if s := task.SlotsNeeded(); (task.fittingRequirements.DedicatedAgent || s > n) && s%n != 0 {
+		if s := task.SlotsNeeded(); s > n && s%n != 0 {
```

Comment:
```suggestion
		if s := task.SlotsNeeded(); s%n != 0 {
```
@aaron276h going to make this change, too, since we would only want to spare `s < n` cases when we were trying to fit multiple commands onto an agent (now covered by the other codepath).